src/assets/scripts/airport/Route/RouteModel.js

```diff
@@ -30,6 +30,8 @@ export default class RouteModel {
      * @param routeString {string}
      */
     constructor(routeString) {
+        super();
+
         if (typeof routeString === 'undefined' || typeof routeString !== 'string') {
```

Comment:
it could, but then I'd be missing a case where I want to exit early.  By checking for both, I provide an easy exit that doesn't instantiate the class _and_ doesn't throw.

If this class were instantiated thusly: `new RouteModel()`, that would give us a `routeString` that is `undefined`.  This will then throw on the next line because `undefined` does not pass the `_isValidRouteString()` method.

Really, I'm just checking that the `routeString` is in fact a string and nothing else.  If it is a string, and the class ends up throwing in the next block, something has gone wrong or something is doing a thing it shouldn't be.  I could have used `lodash._isNil()` here because that checks for both `undefined` and `null`. I chose the native approach because it's small and I am checking explicitly for `undefined`.

Remember, `undefined` is a specific data type in Javascript, just like `NaN` or `null`.